bin/src/utils.dart

```diff
@@ -0,0 +1,265 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/file_system/file_system.dart' as fs;
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:analyzer/source/package_map_resolver.dart';
+import 'package:analyzer/source/pub_package_map_provider.dart';
+import 'package:analyzer/source/sdk_ext.dart';
+import 'package:analyzer/src/dart/sdk/sdk.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/java_io.dart';
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer/src/generated/source_io.dart';
+import 'package:path/path.dart' as path;
+
+final RegExp newLinePartOfRegexp = new RegExp('\npart of ');
+final RegExp partOfRegexp = new RegExp('part of ');
+
+Iterable<ClassElement> getClassesThatExtendFromModule(
+    AnalysisContext context, Directory sdkDir) {
+  final entryPoints = getPackageEntryPoints();
+  final sources = parseSources(context, entryPoints);
+  verifyAnalysis(context, sources);
+
+  return parseLibraries(context, sources)
+      .expand(getSubclassesOfLifecycleModule)
+      .where((element) => element.getGetter('name') == null);
```

Comment:
Would `getField` work here since it takes synthetic fields into account?

If so, it might be advantageous to `getGetter` since it also handles overrides of the form`final name = 'FooModule';`, but I'm not 100% on that.

Reply:
Also #nit based on the method name, this `where` seems to belong with the consumption of the return value and not inside the method itself.